### PR TITLE
Switch to using latest Pulumi versions and remove redundant CLI installation

### PR DESCRIPTION
**Pull request by leezen**

### CODE REVIEW

File: infrastructure/package.json [hunk 1]
```diff
@@ -6,8 +6,9 @@
         "typescript": "^3.5.3"
     },
     "dependencies": {
-        "@pulumi/aws": "^2.0.0",
-        "@pulumi/pulumi": "^2.0.0",
+        "@pulumi/aws": "^4.0.0",
+        "@pulumi/pulumi": "^3.0.0",
+        "hosted-git-info": "2.8.9",
```
Comment (cnunciato) [inline infrastructure/package.json:11]: Was this added intentionally?

**Review by cnunciato — COMMENTED** [review]
I see that we're using the runner-installed Pulumi, but the PR title suggests an "action" related change I'm not seeing. Maybe I'm missing something?

File: infrastructure/package.json (hunk 1 above)
Comment (leezen) [reply infrastructure/package.json:11]: No. I meant to upgrade what was in `yarn.lock` per the dependabot alert. I thought I had run the correct command and didn't see this sneak in. Thanks for pointing out.

**Comment by leezen** [comment]
> I see that we're using the runner-installed Pulumi, but the PR title suggests an "action" related change I'm not seeing. Maybe I'm missing something?

I'll update the description to clarify. What I meant is that the latest action installs the CLI now, so there shouldn't be a need to include the former install action.

**Review by cnunciato — APPROVED** [review]

**Pull request merged**